prody/atomic/atomgroup.py

```diff
@@ -1229,18 +1279,31 @@ def setBonds(self, bonds):
             raise ValueError('bonds.shape must be (n_bonds, 2)')
         if bonds.min() < 0:
             raise ValueError('negative atom indices are not valid')
+
         n_atoms = self._n_atoms
         if bonds.max() >= n_atoms:
             raise ValueError('atom indices are out of range')
+
         bonds.sort(1)
         bonds = bonds[bonds[:, 1].argsort(), ]
         bonds = bonds[bonds[:, 0].argsort(), ]
         bonds = np.unique(bonds, axis=0)
 
+        d = {}
+        for n, b in enumerate(bonds):
+            key = '%d %d'%(b[0], b[1])
+            d[key] = n
+        self._bondIndex = d
+
         self._bmap, self._data['numbonds'] = evalBonds(bonds, n_atoms)
         self._bonds = bonds
         self._fragments = None
 
+        if bondOrders is None:
+            self._bondOrders = None
```

Comment:
with the current code it would not work as we would do self._bondOrders = np.array(bondOrders, np.int8) in setBondOrders with bondOrders==None. Howerver, I agree that logicall this shoudl be handled in setBondOrders so I moved the if block from setBonds() to setBondOrders() allowing to call setBondOrders(None) to "forget" this info

Reply:
Sounds good